gdal/gcore/overview.cpp

```diff
@@ -2757,6 +2809,48 @@ GDALResampleChunkC32R( int nSrcWidth, int nSrcHeight,
                         static_cast<float>(dfRatio);
                 }
             }
+            else if( EQUAL(pszResampling, "RMS") )
+            {
+                double dfTotalR = 0.0;
+                double dfTotalI = 0.0;
+                double dfR = 0.0;
+                double dfI = 0.0;
+                int nCount = 0;
+
+                for( int iY = nSrcYOff; iY < nSrcYOff2; ++iY )
+                {
+                    for( int iX = nSrcXOff; iX < nSrcXOff2; ++iX )
+                    {
+                        dfR = pafSrcScanline[iX*2+static_cast<GPtrDiff_t>(iY-nSrcYOff)*nSrcWidth*2];
+                        dfI = pafSrcScanline[iX*2+static_cast<GPtrDiff_t>(iY-nSrcYOff)*nSrcWidth*2+1];
+
+                        // TODO(schwehr): Maybe use std::complex?
+                        dfTotalR += dfR * dfR - dfI *dfI;
+                        dfTotalI += 2. * dfR * dfI;
+
+                        ++nCount;
+                    }
+                }
+
+                CPLAssert( nCount > 0 );
+                if( nCount == 0 )
+                {
+                    pafDstScanline[iDstPixel*2] = 0.0;
+                    pafDstScanline[iDstPixel*2+1] = 0.0;
+                }
+                else
+                {
+                    /* compute mean */
+                    dfTotalR = dfTotalR/nCount;
+                    dfTotalI = dfTotalI/nCount;
+                    /* compute square root of the complex mean */
+                    dfR = sqrt((dfTotalR + sqrt(dfTotalR * dfTotalR + dfTotalI * dfTotalI))/2.);
+                    dfI = dfTotalI / fabs(dfTotalI) * sqrt((-dfTotalR + sqrt(dfTotalR * dfTotalR + dfTotalI * dfTotalI)) / 2.);
```

Comment:
@rouault 
computation of RMS for complex should be the same as gdal/alg/gdalwarpkernel.cpp here, should it?

Reply:
yes, we should use the same method in both cases.

Reply:
Done